central/processlisteningonport/store/postgres/full_store.go

```diff
@@ -0,0 +1,167 @@
+package postgres
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	metrics "github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/central/processlisteningonport/store"
+	"github.com/stackrox/rox/generated/storage"
+	ops "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+)
+
+// NewFullStore augments the generated store with GetProcessListeningOnPort functions.
+func NewFullStore(db *pgxpool.Pool) store.Store {
+	return &fullStoreImpl{
+		Store: New(db),
+		db:    db,
+	}
+}
+
+// FullStoreWrap augments the wrapped store with GetProcessListeningOnPort functions.
+func FullStoreWrap(wrapped store.Store) store.Store {
+	return &fullStoreImpl{
+		Store: wrapped,
+	}
+}
+
+type fullStoreImpl struct {
+	Store
+	db *pgxpool.Pool
+}
+
+// NewFullTestStore is used for testing.
+func NewFullTestStore(_ testing.TB, store store.Store) store.Store {
+	return &fullStoreImpl{
+		Store: store,
+	}
+}
+
+// SQL query to join process_listening_on_port together with
+// process_indicators. Used to provide information for queries like 'give
+// me all PLOP by this deployment'.
+// XXX: Verify the query plan to make sure needed indexes are in use.
+const getByDeploymentStmt = "SELECT plop.id, plop.serialized, " +
+	"proc.podid, proc.containername, " +
+	"proc.signal_containerid, " +
+	"proc.signal_name, proc.signal_args, proc.signal_execfilepath, " +
+	"proc.clusterid, proc.serialized as procSerialized " +
+	"FROM process_listening_on_ports plop " +
+	"JOIN process_indicators proc " +
+	"ON plop.processindicatorid = proc.id " +
+	"WHERE proc.deploymentid = $1 AND plop.closed = false"
+
+// Manually written function to get PLOP joined with ProcessIndicators
+func (s *fullStoreImpl) GetProcessListeningOnPort(
+	ctx context.Context,
+	deploymentID string,
+) ([]*storage.ProcessListeningOnPort, error) {
+	defer metrics.SetPostgresOperationDurationTime(
+		time.Now(),
+		ops.GetProcessListeningOnPort,
+		"ProcessListeningOnPortStorage",
+	)
+
+	return pgutils.Retry2(func() ([]*storage.ProcessListeningOnPort, error) {
+		return s.retryableGetPLOP(ctx, deploymentID)
+	})
+}
+
+func (s *fullStoreImpl) retryableGetPLOP(
+	ctx context.Context,
+	deploymentID string,
+) ([]*storage.ProcessListeningOnPort, error) {
+	var rows pgx.Rows
+	var err error
+
+	rows, err = s.db.Query(ctx, getByDeploymentStmt, deploymentID)
+
+	if err != nil {
+		log.Warnf("%s: %s", getByDeploymentStmt, err)
```

Comment:
Should the error processing of the caller function be sufficient to remove this Warn log ?